src/hipscat/pixel_math/partition_stats.py

```diff
@@ -136,18 +136,19 @@ def generate_destination_pixel_map(histogram, pixel_map):
         dictionary that maps the integer 3-tuple of a pixel at destination order to the set of
         indexes in histogram for the pixels at the original healpix order
     """
-
     non_none_elements = [i for i in pixel_map if i is not None]
```

Comment:
I know this isn't part of the change, but I think you could speed this up with Boolean indexing. Something like:

`non_none_elements  = pixel_map[pixel_map != np.array(None)]`

Reply:
That does speed up a little!

Lemme explain this a little:
- generated a (seeded) random histogram at orders [1,2,3,4,5,6,7,8,9,10] (skipped some for the original code because I don't have that kinda time)
- ran the method for each histogram
- horizontal axis is the order of the histogram
- "original" is the current head version
- "refactor" is the first version of the refactor
- "comment" is using your comment
- and the vertical axis is a log scale (because we're talking orders of magnitude speed ups).
- your comment results in a factor of 2 speed up, which doesn't look as impressive on a log-scale, but is definitely worth committing.

![Method execution times (log scale) (1)](https://user-images.githubusercontent.com/113376043/221947350-e0cfd80a-c759-46f8-9aee-353f374ac026.png)